lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/chukwumaibezim/BankAccount.java

```diff
@@ -0,0 +1,138 @@
+package com.codedifferently.lesson7.chukwumaibezim;
+
+import java.util.HashMap;
+import java.util.Map;
+
+// import org.springframework.boot.SpringApplication;
+// import org.springframework.boot.autoconfigure.SpringBootApplication;
+// import org.springframework.context.annotation.Configuration;
+
+// @Configuration
+// @SpringBootApplication(scanBasePackages = "com.chukwumaibezim")
+
+public class BankAccount {
+
+  public static class InsufficientFundsException extends Exception {
+    public InsufficientFundsException(String message) {
+      super(message);
+    }
+  }
+
+  private String accountHolderName;
+  private double balance;
+  private String dateOpened;
+  private String accountType;
+  private int accountNumber;
+
+  public enum AccountType {
+    CHECKING,
+    SAVINGS,
+    CD
+  }
+
+  private Map<Integer, String> additionalInfo = new HashMap<>();
+
+  public BankAccount(
+      String accountHolderName,
+      double balance,
+      String dateOpened,
+      String accountType,
+      int accountNumber) {
+    this.accountHolderName = accountHolderName;
+    this.balance = balance;
+    this.dateOpened = dateOpened;
+    this.accountType = accountType;
+    this.accountNumber = accountNumber;
+  }
+
+  public String getAccountHolderName() {
+    return accountHolderName;
+  }
+
+  public void setAccountHolderName(String accountHolderName) {
+    this.accountHolderName = accountHolderName;
+  }
+
+  public double getBalance() {
+    return balance;
+  }
+
+  public void setBalance(double balance) {
+    this.balance = balance;
+  }
+
+  public String getDateOpened() {
+    return dateOpened;
+  }
+
+  public void setDateOpened(String dateOpened) {
+    this.dateOpened = dateOpened;
+  }
+
+  public String getAccountType() {
+    return accountType;
+  }
+
+  public void setAccountType(String accountType) {
+    this.accountType = accountType;
+  }
+
+  public int getAccountNumber() {
+    return accountNumber;
+  }
+
+  public void setAccountNumber(int accountNumber) {
```

Comment:
nit: I'd expect the account to be initialized with an account number and immutable.

Reply:
updated to final and removed setter